core/src/main/scala/org/apache/spark/scheduler/TaskScheduler.scala

```diff
@@ -121,4 +121,6 @@ private[spark] trait TaskScheduler {
    */
   def applicationAttemptId(): Option[String]
 
+
+  def hasRunningTasks(stageId: Int): Boolean
```

Comment:
Why do we need this method ?

Reply:
[Please see](https://github.com/apache/spark/pull/49270#discussion_r1896703317)
This is the code I used to complete this in the beginning (kill waiting stage which only has running tasks). I'm not sure which way is better.

Reply:
If the method is no longer being used, please remove it.